plugins/transformer/basic/client/recordupdater/record_updater.go

```diff
@@ -186,3 +258,49 @@ func (*RecordUpdater) obfuscateColumn(column arrow.Array) arrow.Array {
 	}
 	return bld.NewStringArray()
 }
+
+func (*RecordUpdater) obfuscateJSONColumn(column arrow.Array, jc jsonColumn) arrow.Array {
+	bld := types.NewJSONBuilder(array.NewExtensionBuilder(memory.DefaultAllocator, types.NewJSONType()))
+	for i := 0; i < column.Len(); i++ {
+		if !column.IsValid(i) {
+			bld.AppendNull()
+			continue
+		}
+
+		val := gjson.Get(column.ValueStr(i), jc.columnPath)
+		if val.Exists() && val.Type == gjson.String {
+			modified, err := sjson.Set(column.ValueStr(i), jc.columnPath, fmt.Sprintf("%x", sha256.Sum256([]byte(val.Str))))
+			if err == nil {
+				bld.AppendBytes([]byte(modified))
+			} else {
+				bld.AppendBytes([]byte(val.Str))
+			}
+			continue
+		}
+
+		bld.AppendBytes([]byte(fmt.Sprintf("%x", sha256.Sum256([]byte(column.ValueStr(i))))))
```

Comment:
Won't this make the JSON value invalid? If the value doesn't exist we probably want to keep the original value as JSON objects can have empty/null fields omitted so it's common to have non existing values

Reply:
Good point, fixed